examples/common_vmseries/variables.tf

```diff
@@ -761,3 +761,137 @@ variable "appgws" {
+    - `address_space`           - (`list(string)`, required when `create_virtual_network = `false`) a list of CIDRs for a newly
+                                  created VNET.
+    - `resource_group_name`     - (`string`, optional, defaults to current RG) a name of an existing Resource Group in which
+                                  the VNET will reside or is sourced from.
+    - `create_subnets`          - (`bool`, optional, defaults to `true`) if `true`, create Subnets inside the Virtual Network,
+                                  otherwise use source existing subnets.
+    - `subnets`                 - (`map`, optional) map of Subnets to create or source, for details see
+                                  [VNET module documentation](../../modules/vnet/README.md#subnets).
+    - `network_security_groups` - (`map`, optional) map of Network Security Groups to create, for details see
+                                  [VNET module documentation](../../modules/vnet/README.md#network_security_groups).
+    - `route_tables`            - (`map`, optional) map of Route Tables to create, for details see
+                                  [VNET module documentation](../../modules/vnet/README.md#route_tables).
+
+    For all properties and their default values see [VNET module documentation](../../modules/vnet/README.md).
+  
+  - `authentication`         - (`map`, optional, defaults to example defaults) authentication settings for the deployed VMs.
+  - `spoke_vms`              - (`map`, required) a map defining test VMs.
+    
+    The most basic properties are as follows:
+    - `name`       - (`string`, required) a name of the VM.
+    - `vnet_key`   - (`string`, required) a key describing a VNET defined in `vnets` property.
+    - `subnet_key` - (`string`, required) a key describing a Subnet found in a VNET definition.
+
+    For all properties and their default values see [module's documentation](../../modules/test_infrastructure/README.md#test_vms).
+  
+  - `bastions`               - (`map`, required) a map containing Azure Bastion definitions.
+    
+    The most basic properties are as follows:
+    - `name`       - (`string`, required) an Azure Bastion name.
+    - `vnet_key`   - (`string`, required) a key describing a VNET defined in `vnets` property. This VNET should already have an
+                     existing subnet called `AzureBastionSubnet` (the name is hardcoded by Microsoft).
+    - `subnet_key` - (`string`, required) a key pointing to a Subnet dedicated to a Bastion deployment.
+
+    For all properties and their default values see [module's documentation](../../modules/test_infrastructure/README.md#bastions).
+
+  EOF
+  default     = {}
+  nullable    = false
+  type = map(object({
+    create_resource_group = optional(bool, true)
+    resource_group_name   = optional(string)
+    vnets = map(object({
+      name                    = string
+      resource_group_name     = optional(string)
```

Comment:
Do we need separate `resource_group_name` both in the root level and vnets level ?

Reply:
It adds flexibility that allows you to create the test environment in pre-existing VNET or create a separate VNET for test infrastructure only. Imo it should stay.

Reply:
Got it but can't we achieve the same just by having the resource_group_name at the object level?

Reply:
Ahh yes, my bad. I thought you mean like global resource_group_name variable. I'll fix that.